Touhou Launcher/MainForm.cs

```diff
@@ -1039,25 +1039,31 @@ private void chkRandom_CheckedChanged(object sender, EventArgs e)
 
         private void browseFile_Click(object sender, EventArgs e)
         {
-            TextBox txtbox = (TextBox)launcherSettings.Controls.Find(((Button)sender).Name.Substring(6).ToLower() + "Dir", false).FirstOrDefault(n => n.GetType() == typeof(TextBox));
-            foreach (string file in MainForm.FileBrowser(this, MainForm.rm.GetString(((Button)sender).Name.Substring(6).ToLower() + "SelectTitle"), MainForm.rm.GetString("executableFilter") + " (*.exe, *.bat, *.lnk)|*.exe;*.bat;*.lnk|" + MainForm.rm.GetString("allFilter") + " (*.*)|*.*"))
+            string controlName = ((Button)sender).Name.Substring(6).ToLower() + "Dir";
+            FieldInfo field = curCfg.GetType().GetField(controlName);
```

Comment:
Could just be my brainworms speaking, but wouldn't it be better to use `typeof(Configs)` here? `curCfg`'s type isn't going to change at this point, so having the type at compile time is preferable. Also, having the program throw an easily noticeable exception in case of a type mismatch is preferable to failing silently.

Reply:
Yeah, that makes sense.